### PR TITLE
User/avneetgill/sft 95/port over code for lights board

### PR DESCRIPTION
**Pull request by badeniran**
Reopening because Baba is dumb

### CODE REVIEW

File: EpsilonLights/Core/Src/Lights.c [hunk 1]
```diff
@@ -38,19 +38,16 @@ void updateLightsTask(void const* arg)
         /* UPDATE HEADLIGHTS */
         if ((headlightsOff))
         {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_OFF);
+            HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, LIGHT_OFF);
         }
         else if ((headlightsLow && headlightsHigh))
         {
             // Error state, turn only the low headlights on.
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:45]: This comment isn't relevant anymore.

File: EpsilonLights/Core/Src/Lights.c [hunk 2]
```diff
@@ -163,7 +160,7 @@ void blinkSignalLightsTask(void const* arg)
     }
 }
 
-void updateStrobeLight(void const* arg)
+void updatSTROBELight(void const* arg)
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:163]: uhhh I don't think strobe is an acronym

File: EpsilonLights/Core/Src/Lights.c [hunk 3]
```diff
@@ -261,52 +267,61 @@ void sendHeartbeatTask(void const* arg)
...
     }
 }
 
 // Reimplement weak definition in stm32f4xx_hal_can.c
-void HAL_CAN_RxCpltCallback(CAN_HandleTypeDef* hcan)
+void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
 {
-    CanRxMsgTypeDef* msg = hcan->pRxMsg;
+    CAN_RxHeaderTypeDef hdr;
+    uint8_t Data[8] = {0};
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:293]: `Data`->`data`

File: EpsilonLights/Core/Src/main.c [hunk 4]
```diff
@@ -272,7 +285,7 @@ static void MX_GPIO_Init(void)
     HAL_GPIO_WritePin(GPIOA, LED_RED_Pin | LED_GREEN_Pin | LED_BLUE_Pin, GPIO_PIN_RESET);
 
     /*Configure GPIO pin Output Level */
-    HAL_GPIO_WritePin(GPIOB, RSIGNAL_Pin | LSIGNAL_Pin | BREAK_Pin | HEAD_Pin
+    HAL_GPIO_WritePin(GPIOB, RSIGNAL_Pin | LSIGNAL_Pin | BRAKE_Pin | HEAD_Pin
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/main.c:288]: lol

File: EpsilonLights/Core/Src/Lights.c [hunk 5]
```diff
@@ -38,19 +38,16 @@ void updateLightsTask(void const* arg)
         /* UPDATE HEADLIGHTS */
         if ((headlightsOff))
         {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_OFF);
+            HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, LIGHT_OFF);
         }
         else if ((headlightsLow && headlightsHigh))
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:43]: Is there still a `headLightsHigh` input? If not I think we should remove all refs to it entirely, rename `headlightsLow`->`headlights` and then make a future task to remove it from the driver controls.

File: EpsilonLights/Core/Src/Lights.c [hunk 6]
```diff
@@ -40,14 +38,14 @@ void updateLightsTask(void const* arg)
...
-        else if ((headlightsLow && headlightsHigh))
+        else if ((headlights))
         {
             // Error state, turn only the low headlights on.
             HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, LIGHT_ON);
         }
         else
         {
-            HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, headlightsLow);
+            HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, headlights);
```
Comment (badeniran) [inline EpsilonLights/Core/Src/Lights.c:48]: I think you can just turn the headlights off in this condition

File: EpsilonLights/Core/Src/Lights.c (hunk 1 above)
Comment (badeniran) [reply EpsilonLights/Core/Src/Lights.c:43]: ^

File: EpsilonLights/Core/Src/Lights.c (hunk 6 above)
Comment (bill-luu) [reply EpsilonLights/Core/Src/Lights.c:48]: I think you can do
```
if(headlightsOff)
{...}
else
{
HAL_GPIO_WritePin(HEAD_GPIO_Port, HEAD_Pin, headlights);
}
```

File: EpsilonLights/Core/Src/Lights.c [hunk 7]
```diff
@@ -163,7 +158,7 @@ void blinkSignalLightsTask(void const* arg)
     }
 }
 
-void updateStrobeLight(void const* arg)
+void updatStrobeLight(void const* arg)
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:161]: `updat`

File: EpsilonLights/Core/Src/Lights.c [hunk 8]
```diff
@@ -219,28 +214,37 @@ void reportLightsToCanTask(void const* arg)
...
+        canTxHdr.StdId = LIGHTS_STATUS_STDID;
+        uint8_t data [1];
+        uint32_t mailbox;
         // Initalize to avoid garbage values in [6] anad [7]
         LightsStatus stat = {0};
         // Read lights gpios
-        stat.lowBeams = HAL_GPIO_ReadPin(HLOW_GPIO_Port, HLOW_Pin);
-        stat.highBeams = HAL_GPIO_ReadPin(HHIGH_GPIO_Port, HHIGH_Pin);
+        stat.lowBeams = HAL_GPIO_ReadPin(HEAD_GPIO_Port, HEAD_Pin);
+        stat.highBeams = 0;
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/Lights.c:234]: hmmmm, at some point the LightStatus should no longer be reporting any `highBeams` fields and just have a headlights one. New issue? Given the changes we need to make elsewhere.

File: EpsilonLights/Core/Src/main.c [hunk 9]
```diff
@@ -321,32 +334,31 @@ static void MX_CAN_UserInit(void)
     batteryFilterConfig.FilterMaskIdLow = 0; //unused
     batteryFilterConfig.FilterFIFOAssignment = 0;
     batteryFilterConfig.FilterActivation = ENABLE;
-    batteryFilterConfig.BankNumber = 0; // Set all filter banks for CAN2
+    batteryFilterConfig.SlaveStartFilterBank = 0; // Set all filter banks for CAN2
```
Comment (bill-luu) [inline EpsilonLights/Core/Src/main.c:337]: I'm not sure what `SlaveStartFilterBank` does. Is the comment still correct?

File: EpsilonLights/Core/Src/Lights.c (hunk 8 above)
Comment (avneetgill) [reply EpsilonLights/Core/Src/Lights.c:234]: Yeah I'll make an issue for it in JIRA and then we can update this and driver controls

File: EpsilonLights/Core/Src/main.c (hunk 9 above)
Comment (avneetgill) [reply EpsilonLights/Core/Src/main.c:337]: I think what it does is that it assigns all of the lower index filter banks to the master CAN instance and the higher ones to the slave CAN instance. It's only used for dual CAN instances.

Comment (badeniran) [reply EpsilonLights/Core/Src/main.c:337]: When I was looking at the new API, I'm pretty sure they just refactored `BankNumber` to be `SlaveStartFilterBank`

**Pull request merged**